common/lib/common-definitions/src/logger.ts

```diff
@@ -29,18 +29,39 @@ export interface ITaggedTelemetryPropertyType {
 
 /**
  * JSON-serializable properties, which will be logged with telemetry.
+ * @deprecated interface will be repurposed to support flat arrays and objects for FF internal logging APIs
  */
 export interface ITelemetryProperties {
     [index: string]: TelemetryEventPropertyType | ITaggedTelemetryPropertyType;
 }
 
+/**
+ * Set of properties defined for the base logger to use only primitive types.
+ */
+// eslint-disable-next-line @rushstack/no-new-null
+export type TelemetryBaseEventPropertyValue = string | number | boolean | null | undefined;
+
+/**
+ * A property to be logged to telemetry containing both the value and a tag. Tags are generic strings that can be used
+ * to mark pieces of information that should be organized or handled differently by loggers in various first or third
+ * party scenarios. For example, tags are used to mark sensitive information that should not be stored in logs.
+ */
+export interface ITaggedBaseTelemetryPropertyValue {
+    value: TelemetryBaseEventPropertyValue;
+    tag: string;
+}
+
+export interface ITelemetryBaseProperties {
+    [index: string]: TelemetryBaseEventPropertyValue | ITaggedBaseTelemetryPropertyValue;
+}
+
 /**
  * Base interface for logging telemetry statements.
  * Can contain any number of properties that get serialized as json payload.
  * @param category - category of the event, like "error", "performance", "generic", etc.
  * @param eventName - name of the event.
  */
-export interface ITelemetryBaseEvent extends ITelemetryProperties {
+export interface ITelemetryBaseEvent extends ITelemetryBaseProperties {
```

Comment:
this vastly changes our ITelemetryBaseLogger which is supposed to be very simple, so implementing a logger in trivial, and internally we convert to a more complex type in the telemetry utils package. I'd much rather see us push complexity to that layer, than lift to this layer

Reply:
I believe base logger layer stays exactly the same as it was. Difference is only in names of interfaces - their content is exactly same as old interfaces. At least that's the intention of these changes, I believe.